test/cutting/cut_finding/test_best_first_search.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
```

Comment:
The tests look great, thank you for all this hard work.

I wonder why you didn't want to use the unittest framework we used everywhere else in the cutting tests for the tests in this package?

Reply:
I don't actually have a great reason for that :), perhaps the only thing I can say is that these tests were initially developed on my private cco repo and then ported over.